### PR TITLE
build(docker): add docker image creation configs

### PR DESCRIPTION
**Pull request by komi-eulisa**
- include: add a docker-compose file
- include: add instructions on how to run docker container

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+COPY ${DISTRIBUTION_PATH}/standalone/bin/ ${APP_FOLDER}/bin/
+COPY ${DISTRIBUTION_PATH}/standalone/lib/ ${APP_FOLDER}/lib/
+COPY ${DISTRIBUTION_PATH}/standalone/start.sh ${APP_FOLDER}/
+
+RUN groupadd --system ${USER_GROUP} \
+    && useradd  --system -s /usr/sbin/nologin -g ${USER_GROUP} ${USERNAME} \
+    && mkdir -p data temp ../logs config \
+    && chown -R ${USERNAME}:${USER_GROUP} ${APP_FOLDER} \
+    && chown -R ${USERNAME}:${USER_GROUP} /logs \
+    && rm config/connector.properties
```
Comment (zugazagoitia-eulisa) [inline Dockerfile]: Why is this needed?

File: README.md [hunk 2]
```diff
@@ -1 +1,173 @@
...
+```
+
+### Already built image
+
+A built image of the connector is available here: [here](https://scm.ecodex.eu[]).
+
+Pull the image by running:
+
+```shell
+docker pull https://scm.ecodex.eu/connector:latest
```
Comment (zugazagoitia-eulisa) [inline README.md]: Let's review these coordinates before merging

File: README.md [hunk 3]
```diff
@@ -1 +1,173 @@
...
+* get the latest built image from our JFrog repository
+* build your own image from source code
+
+```
+The following section assumes that you are familiar with Docker and have it installed on your development machine. If not, proceed to install it.
+```
+
+### Already built image
+
+A built image of the connector is available here: [here](https://scm.ecodex.eu[]).
```
Comment (zugazagoitia-eulisa) [inline README.md]: Check link before merging

File: README.md [hunk 4]
```diff
@@ -1 +1,173 @@
...
+### Build your own image
+
+By default, there is a ```Dockerfile``` in the root of this project.
+
+```Before building the image, we assume you have a working development environment.```
+
+Build the connector project in production mode by running:
+
+```shell
+mvn build install -Pproduction
```
Comment (zugazagoitia-eulisa) [inline README.md]: `install`???

File: README.md [hunk 5]
```diff
@@ -1 +1,173 @@
...
+```
+
+Build the connector image by running:
+
+```shell
+docker build -t connector:ltest .
```
Comment (zugazagoitia-eulisa) [inline README.md]:
```suggestion
docker build -t connector:latest .
```

File: README.md [hunk 6]
```diff
@@ -1 +1,173 @@
...
+
+Follow the steps bellow:
+* create a `container` folder
+* copy and paste the `docker-compose.yml` file available at the root of the connector source code into the `container` folder.
+* in addition to the `data`, `config`, and `logs` subfolders and their associated configurations ([see previous section](#connector-configuration)), add a `sql` folder.
+* depending on the DBMS (MySQL in our case) copy and paste the DBMS initialisation scripts into the `sql` subfolder:
+```
+  /domibusConnectorDistribution/target/domibusConnector/documentation/database-scripts/initial/MySQL
+```
+* in the `sql` subfolder rename the files as follows:
```
Comment (zugazagoitia-eulisa) [inline README.md]: Can we rename the sources of manually created SQL scripts to `0_` `1_` and `2_` ?

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+            MYSQL_USER: ecodex
+            MYSQL_PASSWORD: ecodex
+            MYSQL_ROOT_PASSWORD: ecodex
+        ports:
+            - "3307:3306"
+        expose:
+            - "3306"
+        volumes:
+            # based on your DBMS, initial scripts can be found in documentation/database-scripts/initial
+            # copy them and renamed them sequentially and placed them in sql folder
```
Comment (zugazagoitia-eulisa) [inline docker-compose.yml]: Reflect the change here if files are renamed in source

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+            - ./sql/:/docker-entrypoint-initdb.d/
+    connector:
+         # uncomment bellow section if you want to build you connector image during the launch of docker-compose
+         # build:
+         #   context: ./
+         #   dockerfile: Dockerfile
+        container_name: connector
+        image: connector:latest
```
Comment (zugazagoitia-eulisa) [inline docker-compose.yml:28]: Maybe we should reference the JFROG image directly?

File: domibusConnectorBuilder/domibusConnectorBuilderStandalone/src/main/docker/Dockerfile
Comment (zugazagoitia-eulisa) [inline domibusConnectorBuilder/domibusConnectorBuilderStandalone/src/main/docker/Dockerfile:1]: If we're not using it anymore, maybe we should remove it. The java version is also wrong

File: pom.xml [hunk 9]
```diff
@@ -858,7 +859,18 @@
...
+                <!-- spring boot -->
+                <plugin>
+                    <groupId>org.springframework.boot</groupId>
+                    <artifactId>spring-boot-maven-plugin</artifactId>
+                    <version>${spring.boot.version}</version>
+                </plugin>
                 <!-- other -->
+                <plugin>
+                    <groupId>com.google.cloud.tools</groupId>
+                    <artifactId>jib-maven-plugin</artifactId>
```
Comment (zugazagoitia-eulisa) [inline pom.xml]: Can we remove it?

**Review by zugazagoitia-eulisa — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (komi-eulisa) [reply Dockerfile]: fixed

File: README.md (hunk 2 above)
Comment (komi-eulisa) [reply README.md]: Indeed

File: README.md (hunk 3 above)
Comment (komi-eulisa) [reply README.md]: Indeed

File: README.md (hunk 4 above)
Comment (komi-eulisa) [reply README.md]: Fixed

File: README.md (hunk 5 above)
Comment (komi-eulisa) [reply README.md]: Fixed

File: README.md (hunk 6 above)
Comment (komi-eulisa) [reply README.md]: is this OK ?
  * `mysql_initial.sql` -> `0_mysql_initial.sql`
  * `mysql_initial_data.sql` -> `1_mysql_initial_data.sql`
  * `mysql_quartz.sql` -> `2_mysql_quartz.sql`

File: docker-compose.yml (hunk 7 above)
Comment (komi-eulisa) [reply docker-compose.yml]: Done

File: pom.xml (hunk 9 above)
Comment (komi-eulisa) [reply pom.xml]: Done

File: domibusConnectorBuilder/domibusConnectorBuilderStandalone/src/main/docker/Dockerfile
Comment (komi-eulisa) [reply domibusConnectorBuilder/domibusConnectorBuilderStandalone/src/main/docker/Dockerfile:1]: Done

File: README.md (hunk 3 above)
Comment (komi-eulisa) [reply README.md]: Added this: https://scm.ecodex.eu/docker/connector:latest

File: README.md (hunk 6 above)
Comment (komi-eulisa) [reply README.md]: Fixs done by default in the source code. No need to rename the sql files

File: README.md (hunk 2 above)
Comment (komi-eulisa) [reply README.md]: I added this:
https://scm.ecodex.eu/docker/connector:latest

File: docker-compose.yml (hunk 8 above)
Comment (komi-eulisa) [reply docker-compose.yml:28]: For time being, I prefer doing it this way.
may be we can improve it in a near future

File: .github/workflows/docker.yaml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+name: Docker Action
+on:
+  push:
```
Comment (zugazagoitia-eulisa) [inline .github/workflows/docker.yaml:3]: Just on develop for now

**Review by zugazagoitia-eulisa — CHANGES_REQUESTED** [review]
With the workflow change we should be ready to merge

File: .github/workflows/docker.yaml (hunk 10 above)
Comment (zugazagoitia-eulisa) [reply .github/workflows/docker.yaml:3]: Also, the workflow lives in `e-CODEX/workflows` now

Comment (komi-eulisa) [reply .github/workflows/docker.yaml:3]: Done
